kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm-search/src/test/java/org/kie/server/integrationtests/jbpm/search/ProcessSearchServiceIntegrationTest.java

```diff
@@ -0,0 +1,218 @@
+        compareList.put(ProcessInstanceField.CORRELATIONKEY, process.getCorrelationKey());
+        compareList.put(ProcessInstanceField.USER_IDENTITY, USER_YODA);
+        compareList.put(ProcessInstanceField.PARENTPROCESSINSTANCEID, process.getParentId());
+        compareList.put(ProcessInstanceField.STATUS, process.getState());
+        compareList.put(ProcessInstanceField.PROCESSVERSION, process.getProcessVersion());
+        compareList.put(ProcessInstanceField.PROCESSNAME, process.getProcessName());
+
+        List<Long> resultsIds = new ArrayList<>();
+        List<ProcessInstance> results = null;
+
+        results = searchServicesClient.
+                findProcessInstancesWithFilters(createQueryFilterAndEqualsTo(compareList), 0, 100);
+
+        resultsIds = new ArrayList<>();
+        for (ProcessInstance res : results) {
+            resultsIds.add(res.getId());
+        }
+
+        Assertions.assertThat(results).isNotNull();
+        Assertions.assertThat(results.size()).isGreaterThanOrEqualTo(0);
+        Assertions.assertThat(resultsIds).contains(process.getId());
+
+        final ProcessInstance[] instance = new ProcessInstance[1];
+        results.forEach((p) -> {
+            if (p.getId().equals(process.getId())){
+                instance[0] = p;
+            }
+        });
+
+        Assertions.assertThat(instance[0].getContainerId()).isEqualTo(CONTAINER_ID);
+        Assertions.assertThat(instance[0].getId()).isEqualTo(processInstanceId);
+        Assertions.assertThat(instance[0].getProcessName()).isEqualTo(process.getProcessName());
+        Assertions.assertThat(instance[0].getCorrelationKey()).isEqualTo(process.getCorrelationKey());
+        Assertions.assertThat(instance[0].getInitiator()).isEqualTo(USER_YODA);
+        Assertions.assertThat(instance[0].getProcessInstanceDescription()).isEqualTo(process.getProcessInstanceDescription());
+        Assertions.assertThat(instance[0].getParentId()).isEqualTo(process.getParentId());
+        Assertions.assertThat(instance[0].getState()).isEqualTo(process.getState());
+        Assertions.assertThat(instance[0].getProcessVersion()).isEqualTo(process.getProcessVersion());
+        Assertions.assertThat(instance[0].getProcessId()).isEqualTo(process.getProcessId());
+    }
+
+    private void testFindProcessInstanceWithQueryFilter(ProcessInstanceQueryFilterSpec filter, Long processInstanceId) {
+        List<Long> resultsIds = new ArrayList<>();
+        List<ProcessInstance> results = null;
+
+        results = searchServicesClient.
+                findProcessInstancesWithFilters(filter, 0, 100);
+
+        resultsIds = new ArrayList<>();
+        for (ProcessInstance res : results) {
+            resultsIds.add(res.getId());
+        }
+
+        Assertions.assertThat(results).isNotNull();
+        Assertions.assertThat(results.size()).isGreaterThanOrEqualTo(1);
```

Comment:
goal here is to check whether there is at least one entry in the results collection. hasSize() doesn't seem to offer this functionality